managers/bosh-manager/DirectorService.js

```diff
@@ -0,0 +1,1112 @@
+            ScheduleManager
+              .getSchedule(this.guid, CONST.JOB.SCHEDULED_BACKUP)
+              .then(schedule => {
+                logger.info(`Backup Job : ${schedule.name} already scheduled for instance : ${this.guid} with interval ${schedule.repeatInterval}`);
+                return;
+              })
+              .catch((error) => {
+                if (typeof error !== errors.NotFound) {
+                  //NotFound is an expected error.
+                  logger.warn('error occurred while fetching schedule for existing job', error);
+                }
+                if (this.service.backup_interval) {
+                  options.repeatInterval = this.service.backup_interval;
+                }
+                logger.info(`Scheduling Backup for instance : ${this.guid} with backup interval of - ${options.repeatInterval}`);
+                //Even if there is an error while fetching backup schedule, trigger backup schedule we would want audit log captured and riemann alert sent
+                return this.serviceFabrikClient.scheduleBackup(options);
+              });
+          } catch (err) {
+            logger.error(`Error occurred while scheduling backup for instance: ${this.guid}. More info:`, err);
+          }
+        } else {
+          logger.info('Scheduled Backup feature not enabled');
+        }
+      });
+  }
+
+  scheduleAutoUpdate() {
+    const options = {
+      instance_id: this.guid,
+      repeatInterval: CONST.SCHEDULE.RANDOM,
+      timeZone: _.get(config, 'scheduler.jobs.service_instance_update.time_zone', 'UTC')
+    };
+    return utils
+      .retry(tries => {
+        logger.info(`+-> ${CONST.ORDINALS[tries]} attempt to schedule auto update for : ${this.guid}`);
+        if (utils.isFeatureEnabled(CONST.FEATURE.SCHEDULED_UPDATE)) {
+          return this
+            .serviceFabrikClient
+            .scheduleUpdate(options)
+            .catch(err => {
+              logger.error(`Error occurred while setting up auto update for : ${this.guid}`, err);
+              throw err;
+            });
+        } else {
+          logger.warn(` Schedule update feature is disabled. Auto update not scheduled for instance : ${this.guid}`);
+        }
+      }, {
+        maxAttempts: 3,
+        minDelay: 1000
+      })
+      .catch(err => logger.error(`Error occurred while scheduling auto-update for instance: ${this.guid} - `, err));
+  }
+
+  static createDirectorService(instanceId, options) {
+    const planId = options.plan_id;
+    const plan = catalog.getPlan(planId);
+    const context = _.get(options, 'context');
+    const directorService = new DirectorService(instanceId, plan);
+    return Promise
+      .try(() => context ? context : directorService.platformContext)
+      .then(context => directorService.assignPlatformManager(DirectorService.getPlatformManager(context.platform)))
+      .return(directorService);
+  }
+
+  static getPlatformManager(platform) {
```

Comment:
This should be removed all together with each manager for each platform type.

Reply:
Platform managers would still remain. SF2.0 managers are for the provisioning use case. Platform managers are for the consuming platform specific use case like SG creation. Cant be removed.